app/services/adapters/jwe_webhook_destination.rb

```diff
@@ -0,0 +1,39 @@
+module Adapters
+  class JweWebhookDestination
+    class ClientRequestError < StandardError
+    end
+
+    def initialize(url:, key:)
+      @url = url
+      @key = key
+    end
+
+    def send_webhook(body:)
+      response = Typhoeus::Request.new(
+        url,
+        method: :post,
+        body: encrypted_body(body)
+      ).run
+
+      unless response.success?
+        raise ClientRequestError, "request for #{url} returned response status of: #{response.code}"
+      end
+    end
+
+    private
+
+    def body_as_string(body)
+      if body.is_a?(String)
+        body
+      else
+        body.to_json
+      end
+    end
+
+    def encrypted_body(body)
+      JWE.encrypt(body_as_string(body), key, alg: 'dir')
```

Comment:
Why are we choosing this encryption algorithm over the others provided? is this part of an adr? what makes this better than the default encryption?

Reply:
the default uses an RSA key. I'm happy to use that instead if preferred? I thought a string would be more portable and easier to setup

Reply:
the ADR does suggest a shared secret

Reply:
I would say that requiring a user (form owner wanting to integrate FB to a 3rd party) to generate and manage certificates is a big ask. A pre-shared string was the preference because of this. In the future we could enforce certain requirements on this string, eg: needs to be 32 chars long.

Reply:
just had a look and you are both right this only select the key type (the encryption is defaulting to `A128CBC-HS256`

the key size required is 16 bytes, so all keys will be 24 chars and look like this `4Ce86rn0TK5tcEF/EtRENA==` (after you base64 it so its readable)